File: Data Structures & Algorithms/binary-tree-right-side-view/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []

        result = [] 
        queue = deque([root])

        while queue:
            l = len(queue)
            last = None
            for index in range(l):
                item = queue.popleft()
                if item:
                    last = item
                    queue.append(item.left)
                    queue.append(item.right)
            if last:
                result.append(last.val)

        return result

File: Data Structures & Algorithms/same-binary-tree/submission-3.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun isSameTree(p: TreeNode?, q: TreeNode?): Boolean {
        val stack = ArrayDeque<Pair<TreeNode?, TreeNode?>>()
        stack.addLast(Pair(p, q))

        while (stack.isNotEmpty()) {
            val (node1, node2) = stack.removeLast()

            if (node1 == null && node2 == null) continue
            if (node1 == null || node2 == null || node1.`val` != node2.`val`) {
                return false
            }
            stack.addLast(Pair(node1.right, node2.right))
            stack.addLast(Pair(node1.left, node2.left))
        }

        return true
    }
}